numbat/src/typechecker.rs

```diff
@@ -487,134 +497,206 @@ impl TypeChecker {
-                                    typed_ast::BinaryOperator::Add => "addition".into(),
-                                    typed_ast::BinaryOperator::Sub => "subtraction".into(),
-                                    typed_ast::BinaryOperator::Mul => "multiplication".into(),
-                                    typed_ast::BinaryOperator::Div => "division".into(),
-                                    typed_ast::BinaryOperator::Power => "exponentiation".into(),
-                                    typed_ast::BinaryOperator::ConvertTo => {
-                                        "unit conversion".into()
-                                    }
-                                    typed_ast::BinaryOperator::LessThan
-                                    | typed_ast::BinaryOperator::GreaterThan
-                                    | typed_ast::BinaryOperator::LessOrEqual
-                                    | typed_ast::BinaryOperator::GreaterOrEqual
-                                    | typed_ast::BinaryOperator::Equal
-                                    | typed_ast::BinaryOperator::NotEqual => "comparison".into(),
-                                    typed_ast::BinaryOperator::LogicalAnd => "and".into(),
-                                    typed_ast::BinaryOperator::LogicalOr => "or".into(),
-                                },
-                                span_expected: lhs.full_span(),
-                                expected_name: " left hand side",
-                                expected_dimensions: self
-                                    .registry
-                                    .get_derived_entry_names_for(&lhs_type),
-                                expected_type: lhs_type,
-                                span_actual: rhs.full_span(),
-                                actual_name: "right hand side",
-                                actual_name_for_fix: "expression on the right hand side",
-                                actual_dimensions: self
-                                    .registry
-                                    .get_derived_entry_names_for(&rhs_type),
-                                actual_type: rhs_type,
-                            },
-                        ))
+                // DateTime types need special handling here, since they're not scalars with dimensions, yet some select binary operators can be applied to them
+                // TODO how to better handle all the operations we want to support with date
+                if lhs_checked.get_type() == Type::DateTime {
+                    let rhs_is_time = dtype(&rhs_checked)
+                        .ok()
+                        .map(|t| t.is_time_dimension())
+                        .unwrap_or(false);
+                    let rhs_is_datetime = rhs_checked.get_type() == Type::DateTime;
+
+                    if *op == BinaryOperator::ConvertTo
+                        && matches!(rhs_checked, typed_ast::Expression::String(..))
```

Comment:
Is there any reason why you implemented it like this, compared to
```suggestion
                        && rhs_checked.get_type() == Type::String
```
which would allow us to do things like:
```py
let tz_germany = "Europe/Berlin"
now() -> tz_germany
```

Reply:
No good reason that I can recall 😃